src/utilities.ts

```diff
@@ -152,93 +158,133 @@ export function getChecksumCommand() {
   return globalCheckSumCommand;
 }
 
-export function loadLaunchFile(jsonPath: string): any {
+export interface jsonDataType {
+  jsonData:any;
+  jsonDataAsString:string;
+}
+
+export function loadLaunchFile(jsonPath: string): jsonDataType|undefined {
+
   // this function takes the path to a launch.json
   // and returns the contents, or an empty list of configurations
   // if we cannot read the file
-  let existingJSON: any;
-  try {
-    existingJSON = JSON.parse(fs.readFileSync(jsonPath));
-  } catch {
-    // if file is empty ...
-    existingJSON = { configurations: [] };
+  let returnValue:jsonDataType|undefined = undefined;
+  
+  // Requires json-c parsing to handle comments etc.
+  const existingContents = fs.readFileSync (jsonPath).toString();
+  var parseErrors: jsonc.ParseError[] = [];  // not using programatically, for debug only
+  // note that jsonc.parse returns "real json" without the comments
+  const existingJSONdata = jsonc.parse(existingContents, parseErrors, jsoncParseOptions);
```

Comment:
Should we make this into a `parseJsonCFile` function? This would avoid the duplication of code such as `var parseErrors` (which happens 3 times).

Reply:
I think that is over-functioning :)  maybe we can just make the parseErrors global since they are not used??

Reply:
The prototype of `parse` has `errors?: {error: ParseErrorCode;}[]` -- can we try passing in `null`?

If we were doing something with `parseErrors`, then fine, but we're not currently.